### PR TITLE
Speedup Travis CI

### LINKED ISSUE #1728: Composer cache on Travis CI
| Q                 | A
| ----------------- | ---
| Issue Type        | Feature Request
| Deployer Version  | N/A
| Local Machine OS  | N/A
| Remote Machine OS | N/A

### Description 
I think it is good to add cache setting into `.travis.yml` for speed up auto test on Travis CI.

It is a setting I think that it is better to add.

```yml
cache:
  directories:
    - $HOME/.composer/cache/files
```

I referenced following doc and article
- https://docs.travis-ci.com/user/caching/#arbitrary-directories
- https://blog.wyrihaximus.net/2015/07/composer-cache-on-travis/

### PR DESCRIPTION
**Pull request by daskepon**
| Q             | A
| ------------- | ---
| Bug fix?      | No
| New feature?  | Yes
| BC breaks?    | No
| Deprecations? | No
| Fixed tickets | #1728 

I want to resolve #1728 . 

- using composer cache on Travis CI.
- using container on Travis CI.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,5 +1,7 @@
 language: php
 
+sudo: false
```
Comment (daskepon) [inline .travis.yml]: Using containers with travis CI.

File: .travis.yml [hunk 2]
```diff
@@ -10,13 +12,18 @@ env:
...
+
+addons:
+  apt:
+    packages:
+      - acl
 
 install:
   - composer self-update
-  - composer update --no-interaction --prefer-source $COMPOSER_FLAGS
+  - composer update --no-interaction --prefer-dist $COMPOSER_FLAGS
```
Comment (daskepon) [inline .travis.yml]: Using composer insatll.

File: .travis.yml (hunk 1 above)
Comment (staabm) [reply .travis.yml]: Afair this is deprecated and will be removed

Comment (antonmedv) [reply .travis.yml]: Hmmm

Comment (staabm) [reply .travis.yml]: See https://blog.travis-ci.com/2018-10-04-combining-linux-infrastructures

Comment (daskepon) [reply .travis.yml]: @staabm Thank you. I did not check this news. I understand.
@antonmedv I will send new PR later.

**Pull request merged**